GNUmakefile

```diff
@@ -0,0 +1,44 @@
+M := .cache/makes
+M-COMMIT ?= 683675d4c45f90f40215399ebd319805877999c9
+$(shell [ -d $M ] || ( \
+  git clone -q https://github.com/makeplus/makes $M && \
+  git -C $M reset -q --hard $(M-COMMIT)))
+ifneq ($(shell git -C $M rev-parse $(M-COMMIT)),\
+       $(shell git -C $M rev-parse HEAD))
+$(error $M is not at the correct commit: $(M-COMMIT))
+endif
+
+include $M/init.mk
+include $M/go.mk
```

Comment:
This installs Go locally, but we shouldn't be doing that in tests as it conflicts with the GitHub Actions where each test run uses a different Go version to ensure compatibility.

It's very uncommon for Go projects to download the runtime to run tests. Why are we doing this?

Reply:
The GNUmakefile #38 is an optional way to do developer testing, which I prefer.

I wasn't intending it for GHA.

We can either wire up the YTS testing directly in GHA, or if we want to use the GNUmakefile to run the YTS tests in GHA, I can make it not install a local go in that context.